vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.ts

```diff
@@ -241,22 +241,26 @@ import { GridFlowSelectionColumn } from "./vaadin-grid-flow-selection-column.js"
         });
 
         grid.$connector.fetchPage = tryCatchWrapper(function (fetch, page, parentKey) {
+          // Adjust the requested page to be within the valid range in case
+          // the grid size has changed while fetchPage was debounced.
+          if (parentKey === root) {
+            page = Math.min(page, Math.floor((grid.size - 1) / grid.pageSize));
```

Comment:
**note:** This line was previously [here](https://github.com/vaadin/flow-components/pull/5897/files#diff-255f48ed31d114d33fd62d5f44b260202e622606b1f6ffb8fa39fbbc96306a84L279). I moved it up just for better readability. The tests also revealed a bug: the previous version always added +1 to the number of the last page, thus giving an invalid page again.